src/main/groovy/edu/oregonstate/mist/videogamesampleapi/resources/PlatformResource.groovy

```diff
@@ -110,4 +111,42 @@ class PlatformResource extends Resource {
             responseBuilder.build()
         }
     }
+
+    @Path("/{id : \\d+}")
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response putPlatform(@Valid Platform newPlatform, @PathParam("id") Integer id) {
+        ResponseBuilder responseBuilder
+        Platform returnPlatform
+
+        Platform checkPlatform = platformDAO.getPlatformById(id)
+
+        //POSTs if the platform doesn't already exist
+        if (checkPlatform == null) {
+            try {
+                platformDAO.postPlatform(newPlatform.getReleaseYear(), newPlatform.getName(), newPlatform.getManufacturer(), newPlatform.getComputer(), newPlatform.getConsole())
+                returnPlatform = platformDAO.platformByName(newPlatform.getName())
+                responseBuilder = ok(returnPlatform)
+                responseBuilder.build()
+            } catch (UnableToExecuteStatementException err) {
+                String executeError = err.getMessage()
+                responseBuilder = badRequest(executeError)
+                responseBuilder.build()
+            }
+        }
+        //Otherwise execute a PUT
+        else {
```

Comment:
The else should be in the previous line with the previous brace. Please create a ticket so that we can add a rule to codenarc to handle this.